tests/tracing/test_provider.py

```diff
@@ -112,3 +118,56 @@ def foo():
     # Re-enable tracing
     mlflow.tracing.enable()
     assert _is_enabled()
+
+
+@pytest.mark.parametrize("enable_mlflow_tracing", [True, False])
+def test_enable_mlflow_tracing_switch_in_serving_fluent(monkeypatch, enable_mlflow_tracing):
+    monkeypatch.setenv("ENABLE_MLFLOW_TRACING", str(enable_mlflow_tracing).lower())
+    monkeypatch.setenv("IS_IN_DB_MODEL_SERVING_ENV", "true")
+
+    @mlflow.trace
+    def foo():
+        return 1
+
+    request_ids = ["id1", "id2", "id3"]
+    with mock.patch(
+        "mlflow.tracing.processor.inference_table.maybe_get_request_id", side_effect=request_ids
```

Comment:
```suggestion
        "mlflow.tracing.processor.inference_table.maybe_get_request_id", return_value=request_ids
```

Reply:
btw why do we need check request IDs?

Reply:
inference_table needs request_id to process spans, using side_effect because it's called three times and I think return_value is not correct

Reply:
Thanks for the explanation! You're right :)